### PR TITLE
PatchValue form

### PR DESCRIPTION
**Pull request by jantilburgh**
Added the possibility to set a "formUpdateStrategy", this will make it possible to partially update a form when not all data is available.

WARNING: this PR also contains feature module support.

### CODE REVIEW

File: src/lib/binding.service.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
 @Injectable()
 export class BindingService {
-  private bindings: {[k: string]: StoreFormBinding} = {};
+  private bindings: { [k: string]: StoreFormBinding } = {};
 
-  constructor(@Inject(STORE_FORMS_CONFIG) private config: StoreFormsConfig,
-              @Optional() private store: Store<any>) {
+  constructor( @Inject(STORE_FORMS_CONFIG) private config: StoreFormsConfig,
+    @Optional() private store: Store<any>) {
```
Comment (gionkunz) [inline src/lib/binding.service.ts:16]: We should avoid such format changes. What ide are you using?

File: src/lib/binding.service.ts [hunk 2]
```diff
@@ -22,46 +22,55 @@ export class BindingService {
   bind(path: string, formGroup: FormGroup) {
     const formGroupSubscription = formGroup.statusChanges
       .pipe(
-        startWith(true)
+      startWith(true)
```
Comment (gionkunz) [inline src/lib/binding.service.ts:25]: Same formating issues here, we'll run into constant conflicts if we are not consistent here

File: src/lib/store-forms.module.ts [hunk 3]
```diff
@@ -33,17 +33,31 @@ export const defaultStoreFormsConfig: StoreFormsConfig = {
   }]
 })
 export class StoreFormsModule {
+  static forFeature(feature: string, config?: StoreFormsConfig) {
+    if (!!!config) {
+      config = <StoreFormsConfig>{
+        feature: feature
+      };
+    } else {
+      config = Object.assign(config, { feature: feature });
```
Comment (gionkunz) [inline src/lib/store-forms.module.ts:42]: Use object property destructuring and property shorthand: `config = {...config, {feature}}`

File: src/lib/store-forms.module.ts [hunk 4]
```diff
@@ -33,17 +33,31 @@ export const defaultStoreFormsConfig: StoreFormsConfig = {
...
+    }
+
+    return this.configure(config);
+  }
   static configure(config?: StoreFormsConfig): ModuleWithProviders {
```
Comment (gionkunz) [inline src/lib/store-forms.module.ts:47]: If we use the `forFeature` convention, we should probably rename this to `forRoot` ?

**Review by gionkunz — APPROVED** [review]

**Comment by gionkunz** [comment]
I've added a refactored version of your changes in version 0.0.4

File: src/lib/binding.service.ts (hunk 1 above)
Comment (jantilburgh) [reply src/lib/binding.service.ts:16]: VScode

**Pull request closed without merging**